config/config.go

```diff
@@ -447,8 +449,46 @@ type ScrapeConfig struct {
 
 	// List of target relabel configurations.
 	RelabelConfigs []*relabel.Config `yaml:"relabel_configs,omitempty"`
+
 	// List of metric relabel configurations.
 	MetricRelabelConfigs []*relabel.Config `yaml:"metric_relabel_configs,omitempty"`
+
+	// List of rules to execute at scrape time.
+	RuleConfigs []*ScrapeRuleConfig `yaml:"rule_configs,omitempty"`
```

Comment:
What are the semantics here in regards to `MetricRelabelConfig` and `RelabelConfig`. Can we document it maybe here?

Essentially what's the order of those relabellings.

I wonder if decision to put rule configs here will require another relabel config - so we can do relabelling before scrape, after scrape and after recording rules.

Some use cases:
A. I want to drop some series/labels on scrape or rename them, before recording
B. I want to drop some series after recording, so I have only recorded series.

Another alternative is to bring rule configuration into Metric relabelling 🤔 (probably less intuitive and harder to use) or do Rule recording BEFORE MetricRelabelConfig if we don't see clear use case like `A` - it might be arguable.

WDYT? cc @roidelapluie @prymitive

Reply:
On second thought, I think we can allow applying metric relabelings to series from scrape rules. The order of operations would then be:
* relabel
* evaluate scrape rules
* metrics relabel

This will allow for use cases where expensive series are aggregated into a record, and then during metric relabeling the original series are dropped and the record is renamed to have the name of the original metric. 

I updated the implementation to allow relabeling recorded samples and documented this behavior in the docs.